src/Compilers/Core/CodeAnalysisTest/AnalyzerAssemblyLoaderTests.cs

```diff
@@ -102,95 +91,106 @@ public AnalyzerAssemblyLoaderTests(ITestOutputHelper testOutputHelper, AssemblyL
-                memberName);
+                state: null,
+                testAction.Method,
+                pathResolvers.NullToEmpty(),
+                assemblyResolvers.NullToEmpty());
 
         private void Run(
             AnalyzerTestKind kind,
             object state,
             Action<AnalyzerAssemblyLoader, AssemblyLoadTestFixture, object> testAction,
-            IAnalyzerAssemblyResolver[]? externalResolvers = null,
-            [CallerMemberName] string? memberName = null) =>
+            ImmutableArray<IAnalyzerPathResolver> pathResolvers = default,
+            ImmutableArray<IAnalyzerAssemblyResolver> assemblyResolvers = default) =>
             Run(
                 kind,
                 state,
-                static (_, _) => { },
                 testAction.Method,
-                externalResolvers,
-                memberName);
+                pathResolvers.NullToEmpty(),
+                assemblyResolvers.NullToEmpty());
 
         private void Run(
             AnalyzerTestKind kind,
-            Action<AssemblyLoadContext, AssemblyLoadTestFixture> prepLoadContextAction,
-            Action<AnalyzerAssemblyLoader, AssemblyLoadTestFixture> testAction,
-            IAnalyzerAssemblyResolver[]? externalResolvers = null,
-            [CallerMemberName] string? memberName = null) =>
-            Run(
+            object? state,
+            MethodInfo method,
+            ImmutableArray<IAnalyzerPathResolver> pathResolvers,
+            ImmutableArray<IAnalyzerAssemblyResolver> assemblyResolvers)
+        {
+            var util = new InvokeUtil();
+            util.Exec(
+                TestOutputHelper,
+                pathResolvers,
+                assemblyResolvers,
+                TestFixture,
                 kind,
-                state: null,
-                prepLoadContextAction,
-                testAction.Method,
-                externalResolvers,
-                memberName);
+                method.DeclaringType!.FullName!,
+                method.Name,
+                state);
+        }
 
         private void Run(
-            AnalyzerTestKind kind,
-            object? state,
-            Action<AssemblyLoadContext, AssemblyLoadTestFixture> prepLoadContextAction,
-            MethodInfo method,
-            IAnalyzerAssemblyResolver[]? externalResolvers,
-            string? memberName)
+            AnalyzerAssemblyLoader loader,
+            Action<AnalyzerAssemblyLoader, AssemblyLoadTestFixture> testAction)
         {
-            var alc = new AssemblyLoadContext($"Test {memberName}", isCollectible: true);
-            try
-            {
-                prepLoadContextAction(alc, TestFixture);
-                var util = new InvokeUtil();
-                util.Exec(TestOutputHelper, alc, TestFixture, kind, method.DeclaringType!.FullName!, method.Name, externalResolvers ?? [], state);
-            }
-            finally
-            {
-                alc.Unload();
```

Comment:
Why was unloading removed from the NET Core utils?

Reply:
The tests used to use three ALC: default, the one created above, and the directory alc. I realized when working on the tests that the middle ALC isn't actually used at all. There was one specific test that was using it but I just moved the ALC creation into that test vs. having it here. I realized that having a blank ALC here was making the tests less representative of real world code.